File: hw/ayuquina/u3/hw29TemplateAndCompsite/HW29CompositePattern/src/main/java/ec/edu/espe/event/hw29compositepattern/Supervisior.java

package ec.edu.espe.event.hw29compositepattern;

import java.util.Vector;

/**
 *
 * @author Danny Ayuquina, LogicLegion, DCCO-ESPE
 */
abstract class Supervisior extends Employee{
    protected Vector directReports = new Vector();
    
    public void stateName(){
        super.stateName();
        if(directReports.size() > 0){
            for(int i=0; i < directReports.size(); ++i){
                ((Employee)directReports.elementAt(i)).stateName();
            }
        }
    }
    
    public void add(Employee anEmployee){
        this.directReports.addElement(anEmployee);
    }
}

File: hw/Vaca/u3/HW29TemplateAndCompsite/HW29CompositePatternJavaScript/Supervisior.js
class Supervisior extends Employee {
    constructor() {
        super();
        this.directReports = [];
    }

    stateName() {
        super.stateName();
        if (this.directReports.length > 0) {
            this.directReports.forEach(employee => employee.stateName());
        }
    }

    add(anEmployee) {
        this.directReports.push(anEmployee);
    }
}
